### PR TITLE
SALTO-1449: Fix router moving annotations to values when wrapping instance additions

### PR DESCRIPTION
**Pull request by ori-moisis**
When instance annotations were added and the router had to wrap them with a new instance
it would move the annotations to the instance values instead

---

This case was triggered by running fetch align and getting a new annotation.
This would create a change where the instance would have this annotation in its `.value` instead of in `.annotations`.
When this was stored in the nacl cache, the issue would persist and would cause trouble when trying to deploy that instance since the rest of the code is not built to handle the case where an annotation is placed in a value (and so, it treated that annotation as if it was part of the value, tried to deploy it, and crashed).

Note that a workaround for existing workspaces is to clear their nacl cache (that would force the parser to read the nacl file, and the parser would place the annotation correctly)

---
_Release Notes_: 
- Fix issue where salesforce metadata deployment would sometimes crash with the error "Maximum call stack size exceeded"

---
_User Notifications_: 
_None_

### CODE REVIEW

File: packages/workspace/test/workspace/nacl_files/elements_cache.test.ts [hunk 1]
```diff
@@ -39,15 +39,15 @@ describe('test cache manager', () => {
     )
   })
 
-  describe('On clear', async () => {
+  describe('On clear', () => {
```
Comment (ori-moisis) [inline packages/workspace/test/workspace/nacl_files/elements_cache.test.ts:42]: unrelated fix, noticed the error message when I ran the tests.
FYI - @yanaise (I believe this was your test)

**Comment by ori-moisis** [comment]
@roironn - can you please take a look so we can merge this fix?

File: packages/workspace/src/workspace/nacl_files/addition_wrapper.ts [hunk 2]
```diff
@@ -94,15 +94,22 @@ const createInstanceElementFromNestedAdditions = (
   path?: string[]
 ): InstanceElement => {
   const value = {}
```
Comment (roironn) [inline packages/workspace/src/workspace/nacl_files/addition_wrapper.ts:96]: I would add a comment here, can be very confusing without context...

**Review by roironn — APPROVED** [review]
Approved with a nit

File: packages/workspace/src/workspace/nacl_files/addition_wrapper.ts (hunk 2 above)
Comment (ori-moisis) [reply packages/workspace/src/workspace/nacl_files/addition_wrapper.ts:96]: what should the comment say?

Comment (roironn) [reply packages/workspace/src/workspace/nacl_files/addition_wrapper.ts:96]: Something about the instance's annotation id abnormality I guess... Even a link to this PR will be enough :)
